render/wlr_renderer.c

```diff
@@ -247,14 +249,29 @@ bool wlr_renderer_init_wl_display(struct wlr_renderer *r,
 	return true;
 }
 
-struct wlr_renderer *wlr_renderer_autocreate(EGLenum platform,
-		void *remote_display) {
-	struct wlr_egl *egl = wlr_egl_create(platform, remote_display, NULL);
+struct wlr_renderer *wlr_renderer_autocreate(struct wlr_backend *backend) {
+	int fd = backend_get_drm_fd(backend);
+	if (fd == 0) {
+		wlr_log(WLR_ERROR, "Invalid DRM fd");
```

Comment:
A better error message might be "Failed to get DRM FD from backend".

Reply:
Makes more sense yeah